latch/utils.py

```diff
@@ -0,0 +1,101 @@
+from typing import Dict
+
+import gql
+import jwt
+from latch_sdk_config.user import user_config
+from latch_sdk_gql.execute import execute
+
+
+def account_id_from_token(token: str) -> str:
+    """Exchanges a valid JWT for a Latch account ID.
+
+    Latch account IDs are needed for any user-specific request, eg. register
+    workflows or copy files to Latch.
+
+    Args:
+        token: JWT
+
+    Returns:
+        A Latch account ID (UUID).
+    """
+    decoded_jwt = jwt.decode(token, options={"verify_signature": False})
+    try:
+        return decoded_jwt.get("id")
+    except KeyError as e:
+        raise ValueError("Your Latch access token is malformed") from e
+
+
+def retrieve_or_login() -> str:
+    """Returns a valid JWT to access Latch, prompting a login flow if needed.
+
+    Returns:
+        A JWT
+    """
+    from latch_cli.services.login import login
+
+    token = user_config.token
+    if token == "":
+        token = login()
+    return token
+
+
+def get_workspaces() -> Dict[str, str]:
+    """Retrieve workspaces that user can access.
+
+    Returns:
+        A dictionary mapping workspace IDs to workspace display names.
+    """
+    account_id = account_id_from_token(retrieve_or_login())
+    res = execute(
+        gql.gql("""
+        query GetUserDefaultWorkspace($accountId: BigInt!) {
+            teamInfos(filter: {owner: {accountId: {equalTo: $accountId}}}) {
+                nodes {
+                    displayName
+                    accountId
+                }
+            }
+            teamMembers(filter: {user: {accountId: {equalTo: $accountId}}}) {
+                nodes {
+                    team {
+                        accountId
+                        displayName
+                    }
+                }
+            }
+        }"""),
+        {"accountId": account_id},
```

Comment:
can u do this using `accountInfoCurrent` somehow instead of passing in `account_id`

Reply:
Unfortunately not.